contracts/axelarnet-gateway/src/msg.rs

```diff
@@ -0,0 +1,60 @@
+use cosmwasm_schema::{cw_serde, QueryResponses};
+use cosmwasm_std::HexBinary;
+use msgs_derive::EnsurePermissions;
+use router_api::{Address, ChainName, CrossChainId, Message};
+
+use crate::state::MessageWithStatus;
+
+#[cw_serde]
+pub struct InstantiateMsg {
+    /// The chain name for this gateway.
+    pub chain_name: ChainName,
+    /// Address of the router contract on axelar.
+    pub router_address: String,
+}
+
+#[cw_serde]
+#[derive(EnsurePermissions)]
+pub enum ExecuteMsg {
+    /// Initiate a cross-chain contract call from Axelarnet to another chain.
+    /// The message will be routed to the destination chain's gateway via the router.
+    #[permission(Any)]
+    CallContract {
+        destination_chain: ChainName,
+        destination_address: Address,
+        payload: HexBinary,
+    },
+
+    /// Forward the given messages to the next step of the routing layer.
+    /// Messages initiated via `CallContract` can be forwarded again to the router.
+    /// If the messages are coming from the router, then they are marked ready for execution.
+    #[permission(Any)]
+    RouteMessages(Vec<Message>),
```

Comment:
It's a bit cumbersome to have to pass the whole message in the rerouting case. I don't know that it makes sense to change the API from the standard gateway API. Who is the primary caller of this? Axelarons?

Reply:
Yeah, I want to conform to the gateway API, since the router also calls this method. Re-routing from source will likely be done manually.

Reply:
So, the current plan is to not conform with the gateway for queries. But I'll keep `RouteMessages` the same since it's a core API for the gateway and the router.